borg/hashindex.pyx

```diff
@@ -238,40 +240,39 @@ cdef class ChunkIndex(IndexBase):
 
     def summarize(self):
         cdef long long size = 0, csize = 0, unique_size = 0, unique_csize = 0, chunks = 0, unique_chunks = 0
-        cdef int32_t *values
+        cdef uint32_t *values
         cdef void *key = NULL
 
         while True:
             key = hashindex_next_key(self.index, key)
             if not key:
                 break
             unique_chunks += 1
-            values = <int32_t*> (key + self.key_size)
+            values = <uint32_t*> (key + self.key_size)
             chunks += _le32toh(values[0])
             unique_size += _le32toh(values[1])
             unique_csize += _le32toh(values[2])
             size += <long long> _le32toh(values[1]) * _le32toh(values[0])
-            csize += <long long> _le32toh(values[2]) *  _le32toh(values[0])
+            csize += <long long> _le32toh(values[2]) * _le32toh(values[0])
 
         return size, csize, unique_size, unique_csize, unique_chunks, chunks
 
     def add(self, key, refs, size, csize):
         assert len(key) == self.key_size
-        cdef int32_t[3] data
+        cdef uint32_t[3] data
         data[0] = _htole32(refs)
         data[1] = _htole32(size)
         data[2] = _htole32(csize)
         self._add(<char*> key, data)
 
-    cdef _add(self, void *key, int32_t *data):
-        cdef int refcount1, refcount2, result
-        values = <int32_t*> hashindex_get(self.index, key)
+    cdef _add(self, void *key, uint32_t *data):
+        cdef uint32_t refcount1, refcount2, result
+        values = <uint32_t*> hashindex_get(self.index, key)
         if values:
             refcount1 = _le32toh(values[0])
             refcount2 = _le32toh(data[0])
-            if (refcount1 < 0 or refcount2 < 0 or
-                    INT32_MAX - refcount1 < refcount2 or INT32_MAX - refcount2 < refcount1):
-                result = INT32_MAX
+            if _MAX_REF - refcount1 < refcount2 or _MAX_REF - refcount2 < refcount1:
```

Comment:
I wonder if a much more readable check here would be:
(unsigned long long) refcount1 + refcount2 > _MAX_REF

This way we nip overflow in the bud during the check and the intent is much more clear too for the reader.